sound/soc/sof/topology.c

```diff
@@ -1299,7 +1299,8 @@ static int sof_widget_load_dai(struct snd_soc_component *scomp, int index,
 	comp_dai.comp.hdr.size = sizeof(comp_dai);
 	comp_dai.comp.hdr.cmd = SOF_IPC_GLB_TPLG_MSG | SOF_IPC_TPLG_COMP_NEW;
 	comp_dai.comp.id = swidget->comp_id;
-	comp_dai.comp.type = SOF_COMP_DAI;
+	comp_dai.comp.type = sdev->pdata->vfe ? SOF_COMP_VIRT_CON :
+		SOF_COMP_DAI;
```

Comment:
@lyakh for my education how does the FW handle this type? Is there a FW PR as well?

Reply:
@ranj063 it doesn't, it doesn't get forwarded to the FW. See `dsp_sof_ipc_tplg_comp_new()` in dsp-sof.c (patch 11/13)